libraries/bot-ai-qna/src/main/java/com/microsoft/bot/ai/qna/dialogs/QnAMakerDialog.java

```diff
@@ -872,13 +877,13 @@ private static void resetOptions(DialogContext dc, QnAMakerDialogOptions dialogO
             // Take this value from GetAnswerResponse
             Boolean isActiveLearningEnabled = response.getActiveLearningEnabled();
 
-            stepContext.setValue(ValueProperty.QNA_DATA, Arrays.asList(response.getAnswers()));
+            stepContext.getValues().put(ValueProperty.QNA_DATA, Arrays.asList(response.getAnswers()));
 
             // Check if active learning is enabled.
             // MaximumScoreForLowScoreVariation is the score above which no need to check
             // for feedback.
-            if (!response.getAnswer().isEmpty() && response.getAnswers()[0]
-                    .getScore() <= (ActiveLearningUtils.getMaximumScoreForLowScoreVariation() / 100)) {
+            if (response.getAnswers().length != 0 && response.getAnswers()[0]
```

Comment:
Are you sure about the change of `!isEmpty()` to `length != 0`?

Reply:
Maybe it is better to do ` length > 0`